x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -50,14 +50,13 @@
     static final Set<String> VALID_WARM_ACTIONS = Sets.newHashSet(ORDERED_VALID_WARM_ACTIONS);
     static final Set<String> VALID_COLD_ACTIONS = Sets.newHashSet(ORDERED_VALID_COLD_ACTIONS);
     static final Set<String> VALID_DELETE_ACTIONS = Sets.newHashSet(ORDERED_VALID_DELETE_ACTIONS);
-    private static final Map<String, Set<String>> ALLOWED_ACTIONS = new HashMap<>();
+    private static final Map<String, Set<String>> ALLOWED_ACTIONS = Map.of(
```

Comment:
This will not compile on backport (which is fine, just wanted to let you know since you asked about this previously)

Reply:
+1, I've pregamed that and I'm ready to do the backport dance.

Reply:
We do have [these utility classes](https://github.com/elastic/elasticsearch/blob/7.x/libs/core/src/main/java/org/elasticsearch/common/collect/Map.java) in the `7.x` branch that provide these functions in a Java8-compatible way so that backports are easier. They're also packaged as a multi-release JAR so that in JVMs that are 9 or later, they use the Java libraries instead, so there's no downside to using them when running with later JVMs.

Reply:
@danhermann the utility classes are unfortunately not that helpful, since to use they the `Map.of` needs to be fully qualified, or `java.util.Map<String, Set<String>>` has to be specified to fully qualify the jdk classes, which I think is more awkward than just changing on backport.